docs/plugin-transform-typescript.md

```diff
@@ -281,6 +186,101 @@ equivalents in Babel can be enabled by some configuration options or plugins.
 - `--watch`, `-w`
   When using `@babel/cli`, you can specify the [`--watch` option](https://babeljs.io/docs/en/babel-cli#compile-files).
 
+## Caveats
+
+Because there are features of the TypeScript language which rely on the full type-system to be available to make changes at runtime. This section of caveats is quite long, however, it's worth noting that a few of these features are only found in older TypeScript codebases and have modern JavaScript equivalents which you are probably already using.
+
+1. Since Babel does not type-check, code which is syntactically correct, but would fail the TypeScript type-checking may successfully get transformed, and often in unexpected or invalid ways.
+
+1. This plugin does not support [`const enum`][const_enum]s because those require type information to compile.
+
+   **Workarounds**:
+
+   - Use the plugin [babel-plugin-const-enum](https://www.npmjs.com/package/babel-plugin-const-enum).
+   - Remove the `const`, which makes it available at runtime.
+
+1. This plugin does not support [`export =`][exin] and [`import =`][exin], because those cannot be compiled to ES.next. These are a TypeScript only form of `import`/`export`.
+
+   **Workarounds**:
+
+   - Use the plugin [babel-plugin-replace-ts-export-assignment](https://www.npmjs.com/package/babel-plugin-replace-ts-export-assignment) to transform `export =`.
+   - Convert to using `export default` and `export const`, and `import x, {y} from "z"`.
+
+1. Changes to your `tsconfig.json` are not reflected in babel. The build process will always behave as though [`isolatedModules`][iso-mods] is turned on, there are Babel-native alternative ways to set a lot of the [`tsconfig.json` options](#typescript-compiler-options) however.
+
+1. **Q**: Why doesn't Babel allow export of a `var` or `let`?
```

Comment:
Not introduced by this PR, but I don't understand this question.

Babel allows exporting anything 🤔